src/Minesweeper.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import reactDom from 'react-dom';
+import PropTypes from 'prop-types';
+import Board from './components/Board/Board';
+import BoardHead from './components/BoardHead/BoardHead';
+
+ class Minesweeper extends React.Component {
+
+    constructor(props){
+        super(props);
+     
+        this.state = {
+                rows: props.rows,
+                columns : props.columns,
+                flags: 0,
+                mines: props.mines,
+                time: 0,
+                status: 0, //at game starts: waiting, running, ended
```

Comment:
What does `0` stand for? use consts instead of magic numbers.
https://en.wikipedia.org/wiki/Magic_number_(programming)
i.e.
```javascript
const BEFORE_GAME_BEGINS = 0
```
and then:
```javascript
status: BEFORE_GAME_BEGINS 
```